Sources/BraintreeThreeDSecure/BTThreeDSecureAuthenticateJWT.swift

```diff
@@ -12,15 +12,14 @@ class BTThreeDSecureAuthenticateJWT {
         forResult lookupResult: BTThreeDSecureResult?,
         completion: @escaping (BTThreeDSecureResult?, Error?) -> Void
     ) {
-        apiClient.sendAnalyticsEvent("ios.three-d-secure.verification-flow.upgrade-payment-method.started")
-
         guard let nonce = lookupResult?.tokenizedCard?.nonce else {
-            apiClient.sendAnalyticsEvent("ios.three-d-secure.verification-flow.upgrade-payment-method.errored")
+            apiClient.sendAnalyticsEvent(BTThreeDSecureAnalytics.jwtAuthFailed)
             completion(nil, BTThreeDSecureError.failedAuthentication("Tokenized card nonce is required."))
             return
         }
 
         guard let urlSafeNonce = nonce.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) else {
+            apiClient.sendAnalyticsEvent(BTThreeDSecureAnalytics.jwtAuthFailed)
```

Comment:
the completion handler from the client (threeDSecureV2Provider?.process) handles general events.

Reply:
I think this is confusing - because now our modules aren't acting the same way. We want to be as consistent as we can across modules.

See my other comment, but I do think in a single follow up PR, we should restructure to go back to the `notify` pattern like in PPCP.

Reply:
The only thing I can think of that might be problem in doing that is that we don't really have a connect right now from client test to for AuthenticateJWT right now.

Reply:
I guess I can test the verifySuccess here.

Reply:
I made a ticket here: https://paypal.atlassian.net/browse/DTBTSDK-2694

Reply:
To threeDSecureV2Provider and to BTThreeDSecureAuthenticateJWT classes, we are not passing merchant completion but a completion that discerns cancel vs other errors and then calls merchant completion. 

The way we have handled it was to just send error or success to that completion from both classes. If we want to handle the errors and analytic messages in the same way, cleanest way is to refactor to pass the merchant completion handler  to both classes.

Reply:
It is a bit tricky with unit testing the general success after authentication since it would require a lot of mock response from Cardinal. I've manually tested general success case from authentication but tough with general fail from authentication. I can look into maybe using Proxyman or some other way to test general fail.